extensions/amp-accordion/1.0/accordion.js

```diff
@@ -321,6 +311,75 @@ export function AccordionSection({
     }
   }, [id, toggleExpanded]);
 
+  const context = useMemo(
+    () => ({
+      animate,
+      contentId,
+      expanded,
+      expandHandler,
+    }),
+    [animate, contentId, expanded, expandHandler]
+  );
+
+  return (
+    <Comp {...rest} expanded={expanded}>
+      <SectionContext.Provider value={context}>
+        {children}
+      </SectionContext.Provider>
+    </Comp>
+  );
+}
+
+/**
+ * @param {!AccordionDef.SectionHeaderProps} props
+ * @return {PreactDef.Renderable}
+ */
+export function AccordionSectionHeader({
+  as: Comp = 'header',
```

Comment:
We need to resolve the conflict between the `<header>`  and `role=button`. If not in this PR, let's add a tracker item.

Reply:
If we need to, we can always wrap the `Comp` in a `button`. This will save us from having to define `tabindex` and `role` as well.

Reply:
Awesome added an item in the tracker to look into this: https://github.com/ampproject/amphtml/issues/30445
Item is called: `Ensure no a11y conflicts from restructure`